CDCgov/prime-reportstream
# 11051: route a message per receiver

This PR updates the route and translate steps such that the route step produces a message per receiver.

Test Steps:
1. Run the smoketests

Confirm that old messages still work:
1. Apply this patch while on `master`
```
index ae81aa9cd..7bca3acfd 100644
--- a/prime-router/src/main/kotlin/azure/QueueAccess.kt
+++ b/prime-router/src/main/kotlin/azure/QueueAccess.kt
@@ -46,7 +46,7 @@ object QueueAccess {
     fun sendMessage(
         queueName: String,
         message: String,
-        invisibleDuration: Duration = Duration.ZERO,
+        invisibleDuration: Duration = Duration.ofSeconds(30),
     ) {
         // Bug:  event.at is calculated before the call to workflowengine.recordHistory
         // In cases of very large datasets, that db write can take a very long time, pushing
```
2. Send an HL7 message through the UP that will get sent to a receiver
3. Immediately after sending the message, turn off reportstream (because of the azure functions bug, I had to go into Activity monitor and kill the java processes)
4. Wait 30 seconds and confirm that there is a message in the convert queue
5. Switch to this branch
6.  Turn report stream on and verify that the report step is delivered
7. Repeat for the route step and translate step (i.e. wait for the message to be in each queue)

## Changes
- The route step now dispatches a message per receiver
- The route step prunes the conditions before uploading the bundle

## Checklist

### Testing

### Process

## Linked Issues
- #11051 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
